### DISCUSSION TITLE
Graphical Extensions

### DISCUSSION THREAD

**Discussion by jeorb** [zed-industries/zed#16836] [Feature Requests]
Will it be possible to create graphical Zed extensions?

Right now, it looks like extension are only for grammars, languages, or themes.

There are a couple of plug-ins or extensions I want to be able to create:

- A Simple Rhai Debugger showing a table of variable values and some controls to step through execution.
- A panel to show some animations controlled by the program being debugged.

The markdown preview in Zed shows some of how these could be done as built in components but I don't see a way to do these as extensions.

I make a toy IDE for an educational game using egui to provide a code editor, debugger, and a graphical panel showing the state as the user's code runs.

This is all pretty easy to make in egui but I'm never going to be able to create a text editor component as good as Zed's.

I think Zed doesn't support this use case yet but what Zed does now is so good that it makes me really want it to be infinitely extendable like neovim, eclipse, and vscode.

**Comment by bajrangCoder** [comment]
Currently it's not possible, But sure in coming future this will be available. As there are many useful gui extension which need to be ported for Zed.
